lib/webauthn/authenticator_response.rb

```diff
@@ -66,11 +75,23 @@ def valid_challenge?(expected_challenge)
       WebAuthn::SecurityUtils.secure_compare(Base64.urlsafe_decode64(client_data.challenge), expected_challenge)
     end
 
-    def valid_origin?(expected_origin)
-      client_data.origin == expected_origin
+    def valid_origin?(rp_id, expected_origin)
+      # If the caller specifies the expected origin, check that it exactly matches the signed origin.
+      return false unless expected_origin.nil? || client_data.origin == expected_origin
```

Comment:
It would be nice to distinguish between the failure cases. Would it make sense to subclass `OriginVerificationError` even further?